### PR TITLE
thumbnail loading with debounce

### PR DESCRIPTION
**Pull request by Harvel218**
| Q               | A
| --------------- | -----
| Bug fix?        | no/yes
| New feature?    | no/yes
| BC breaks?      | no/yes
| Deprecations?   | no/yes 
| Related tickets | fixes #X, partially #Y, mentioned in #Z
| License         | MIT

### CODE REVIEW

File: src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js [hunk 1]
```diff
@@ -47,7 +47,6 @@ export class HandleAutoComplete {
 
         if (mediaContainer.querySelector('input[type=hidden]').value === '') {
```
Comment (tomaszsiwarski) [inline src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js:48]: change it to yoda

File: src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js [hunk 2]
```diff
@@ -47,7 +47,6 @@ export class HandleAutoComplete {
...
             deleteButton.classList.add('is-hidden');
-
             return;
```
Comment (tomaszsiwarski) [inline src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js:50]: there should be empty line before `return;` btw. is `return;` needed?

File: src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js [hunk 3]
```diff
@@ -58,14 +57,21 @@ export class HandleAutoComplete {
...
         mediaContainer.querySelector(this.selectInput).addEventListener('click', (e) => {
             e.preventDefault();
-
             this._getMediaImages(mediaContainer);
         });
+        mediaContainer.querySelector(this.selectInput).addEventListener('input', (e) => {
+            e.preventDefault();
+            clearTimeout(timeout);
+            timeout = setTimeout(() => {
+                console.log('1', e.target.value);
```
Comment (tomaszsiwarski) [inline src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js:69]: :trollface:

File: src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js [hunk 4]
```diff
@@ -58,14 +57,21 @@ export class HandleAutoComplete {
     }
 
     _handleImageChoice(mediaContainer) {
+        let timeout;
```
Comment (tomaszsiwarski) [inline src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js:60]: new line after variable declarations

File: src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js [hunk 5]
```diff
@@ -58,14 +57,21 @@ export class HandleAutoComplete {
...
         mediaContainer.querySelector(this.selectInput).addEventListener('click', (e) => {
             e.preventDefault();
-
             this._getMediaImages(mediaContainer);
         });
+        mediaContainer.querySelector(this.selectInput).addEventListener('input', (e) => {
```
Comment (tomaszsiwarski) [inline src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js:65]: newline before & after this code block (64-64 & 72-73 lines)

File: src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js [hunk 6]
```diff
@@ -58,14 +57,21 @@ export class HandleAutoComplete {
...
+            e.preventDefault();
+            clearTimeout(timeout);
+            timeout = setTimeout(() => {
```
Comment (tomaszsiwarski) [inline src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js:68]: newline before

File: src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js [hunk 7]
```diff
@@ -98,11 +104,12 @@ export class HandleAutoComplete {
...
 
-    async _getMediaImages(mediaContainer) {
+    async _getMediaImages(mediaContainer, value = false) {
         const path = mediaContainer.dataset.bbCmsUrl;
         const typeQuery = mediaContainer.dataset.bbCmsCriteriaType;
-        const url = `${path}&limit=${this.config.limit}&criteria[search][type]=${typeQuery}`;
-
+        const searchValue = value ? `&criteria[search][value]=${value}` : '';
+        const url = `${path}&limit=${this.config.limit}&criteria[search][type]=${typeQuery}&criteria[search][value]=${searchValue}`;
+        console.log(value);
```
Comment (tomaszsiwarski) [inline src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js:112]: :trollface:

File: src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js [hunk 8]
```diff
@@ -98,11 +104,12 @@ export class HandleAutoComplete {
...
     }
 
-    async _getMediaImages(mediaContainer) {
+    async _getMediaImages(mediaContainer, value = false) {
         const path = mediaContainer.dataset.bbCmsUrl;
         const typeQuery = mediaContainer.dataset.bbCmsCriteriaType;
-        const url = `${path}&limit=${this.config.limit}&criteria[search][type]=${typeQuery}`;
-
+        const searchValue = value ? `&criteria[search][value]=${value}` : '';
+        const url = `${path}&limit=${this.config.limit}&criteria[search][type]=${typeQuery}&criteria[search][value]=${searchValue}`;
```
Comment (tomaszsiwarski) [inline src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js:111]: newline after declarations

File: src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js (hunk 3 above)
Comment (Harvel218) [reply src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js:69]: (-:

File: src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js (hunk 2 above)
Comment (Harvel218) [reply src/Resources/assets/admin/js/bitbag/bitbag-media-autocomplete.js:50]: ye, return is needed -

**Pull request merged**